### PR TITLE
AF-1097: [Project Oriented] Endless asset loading appears when discarding unsaved data

### PR DESCRIPTION
**Pull request by paulovmr**
Part of an ensemble:
* https://github.com/kiegroup/appformer/pull/293
* https://github.com/kiegroup/kie-wb-common/pull/1628
* https://github.com/kiegroup/drools-wb/pull/850
* https://github.com/kiegroup/jbpm-designer/pull/768

image

### CODE REVIEW

**Comment by paulovmr** [comment]
jenkins execute full downstream build

**Comment by paulovmr** [comment]
jenkins retest this

**Comment by paulovmr** [comment]
Full downstream build is being ran on the new PR https://github.com/kiegroup/kie-wb-distributions/pull/734

**Comment by barboras7** [comment]
Hi @paulovmr,
I tested the war. I found one issue. When you follow the steps to reproduce as described in the AF-1097 Jira, at the end you get the dialog you just created. If you click "Cancel", you are redirected back to the DRL editor. When you delete the DRL and go back to the team and try to open "test1" project, you get once again the "Unsaved changes" dialog - this should not be the case. 
However, it seems that this only happens with DRL files (the editor is not closed on Delete event) but this issue might be fixed by [this PR](https://github.com/kiegroup/kie-wb-common/pull/1566).

**Comment by paulovmr** [comment]
jenkins execute full downstream build

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/screens/project/close/CloseUnsavedProjectAssetsPopUpPresenter.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        cancelCallback.ifPresent(Command::execute);
+    }
+
+    private String getAssetPath(final WorkspaceProject project,
+                                final String fullPath) {
+        final String projectRootPath = project.getRootPath().toURI();
+        final String relativeAssetPath = fullPath.substring(projectRootPath.length());
+        final String decodedRelativeAssetPath = URIUtil.decode(relativeAssetPath);
+
+        return decodedRelativeAssetPath;
```
Comment (ederign) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/screens/project/close/CloseUnsavedProjectAssetsPopUpPresenter.java]: When cancel, how user comes back to the editor? Did you make a goto to the first asset?

Comment (paulovmr) [reply kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/screens/project/close/CloseUnsavedProjectAssetsPopUpPresenter.java]: Yes, if the cancel action is chosen, the first editor in the popup list will be opened automatically.

**Comment by barboras7** [comment]
Sure, @paulovmr. I'll create a new Jira if the problem is not fixed by Adriel's PR I mentioned above. Thanks for the fixes!

**Comment by paulovmr** [comment]
jenkins execute full downstream build

**Comment by paulovmr** [comment]
jenkins execute full downstream build

**Comment by paulovmr** [comment]
jenkins execute full downstream build

**Comment by paulovmr** [comment]
jenkins execute full downstream build

**Comment by paulovmr** [comment]
jenkins execute full downstream build

**Pull request merged**